rulette-mysql-provider/src/main/java/com/github/kislayverma/rulette/mysql/dao/DataSource.java

```diff
@@ -1,54 +1,49 @@
 package com.github.kislayverma.rulette.mysql.dao;
 
-import com.mchange.v2.c3p0.ComboPooledDataSource;
-import java.beans.PropertyVetoException;
 import java.io.IOException;
 import java.sql.Connection;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.Properties;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
 
 /**
  *
  * @author kislay
  */
-public class DataSource {
+public class DataSource extends HikariDataSource{
```

Comment:
No extends. It encapsulates HikariData Source, but doesn't have to be a Hikari DataSource. We should prefer composition over inheritance Since inheritance fixes the type but composition hides the internal implementation. https://www.w3resource.com/java-tutorial/inheritance-composition-relationship.php